app/activelayer.h

```diff
@@ -0,0 +1,64 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef ACTIVELAYER_H
+#define ACTIVELAYER_H
+
+#include <QObject>
+#include "qgsmaplayer.h"
+#include "qgsvectorlayer.h"
+
+
+/**
+ * @brief The ActiveLayer class holds information about current active layer for recording.
+ * It supports setting active layer by layer and offers information about layer.
+ *
+ * \see LayersProxyModel for methods to convert layer name / index to layer.
+ */
+class ActiveLayer : public QObject
+{
+    Q_OBJECT
+
+    Q_PROPERTY( QString layerId READ layerId )
+    Q_PROPERTY( QString layerName READ layerName )
+    Q_PROPERTY( QgsMapLayer *layer WRITE setActiveLayer READ layer NOTIFY activeLayerChanged )
+    Q_PROPERTY( QgsVectorLayer *vectorLayer READ vectorLayer )
```

Comment:
I think all these properties should have `NOTIFY activeLayerChanged` - otherwise when used in QML, Qt will not know that the value has changed...

Reply:
I put `NOTIFY` only to the one that actually change the active layer - `layer`; others only read the active layer, not changing it so I thought we do not need it there. Is that true or should I add them?

Reply:
As far as I know, if you don't add NOTIFY to a property, QML will think it is constant. So if you use layerName somewhere in QML code and then the active layer changes, then QML will still use the old layer name.

Reply:
updated